docs/src/main/asciidoc/platform.adoc

```diff
@@ -0,0 +1,149 @@
+
+=== Quarkus Platform Artifacts
+
+Each Quarkus platform is defined with a few artifacts.
+
+=== Quarkus Platform BOM
+
+Each Quarkus Platform is expected to provide a Maven BOM artifact that
+
+* imports a chosen version of `io.quarkus:quarkus-bom` (the platform BOM maybe flattened at the end but it has to be based on some version of `io.quarkus:quarkus-bom`)
+* includes all the Quarkus extension artifacts (the runtime and the deployment ones) the platform consists of
+* includes all the necessary thirdparty artifacts that align the transitive dependency versions of the platform extensions to guarantee compatiblity between them
+* includes the <<platform-descriptor,platform JSON descriptor>> artifact
+* possibly includes the <<platform-properties,platform configuration properties>> artifacts
+
+Quarkus applications that want to include extensions from a Quarkus platform will be importing the Quarkus platform BOM.
+
+[[platform-descriptor]]
+=== Quarkus Platform Descriptor
+
+Quarkus platform descriptor is a JSON artifact that provides information about the platform and its extensions to the Quarkus tools. E.g. http://code.quarkus.io and the Quarkus command line tools consult this descriptor to list, add and remove extensions to/from the project on user's request.
+This artifact is also used as a Quarkus platform identifier. When Quarkus tools need to identify the Quarkus platform(s) used in the project, they will analyze the dependency version constraints of the project (the effective list of the managed dependencies from the `dependencyManagement` section in Maven terms) looking for the platform descriptor artifact(s) among them. Given that the platform descriptors are included into the Quarkus platform BOMs, every Quarkus application will inherit the platform descriptor artifact from the imported platform BOM(s) as a dependency version constraint (managed depenency in Maven terms).
+
+To be able to easily identify Quarkus platform descriptors among the project's dependency constraints, the platform descriptor Maven artifact coordinates should follow the folloing naming convention:
+
+* the `groupId` of the descriptor artifact should match the `groupId` of the corresponding Quarkus Platform BOM;
+* the `artifactId` of the descriptor artifact should be the `artifactId` of the corresponding Quarkus Platform BOM with the `-quarkus-platform-descriptor` suffix;
+* the `classifier` of the descriptor artifact should match the `version` of the corresponding Quarkus Platform BOM;
+* the `type` of the descriptor artifact should be `json`;
+* the `version` of the descriptor artifact should match the `version` of the  corresponding Quarkus Platform BOM.
+
+As a string it will look as `<platform-bom-groupId>:<platform-bom-artifactId>-quarkus-platform-descriptor:<platform-version>:json:<platform-version>`
```

Comment:
```suggestion
As a string it will look like `<platform-bom-groupId>:<platform-bom-artifactId>-quarkus-platform-descriptor:<platform-version>:json:<platform-version>`
```